sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,637 @@
+        if self.final_estimator is not None:
+            self.final_estimator_ = clone(self.final_estimator)
+        else:
+            self.final_estimator_ = clone(default)
+
+    @staticmethod
+    def _method_name(name, estimator, method):
+        if estimator in (None, 'drop'):
+            return None
+        if method == 'auto':
+            if getattr(estimator, 'predict_proba', None):
+                return 'predict_proba'
+            elif getattr(estimator, 'decision_function', None):
+                return 'decision_function'
+            else:
+                return 'predict'
+        else:
+            if not hasattr(estimator, method):
+                raise ValueError('Underlying estimator {} does not implement '
+                                 'the method {}.'.format(name, method))
+            return method
+
+    def set_params(self, **params):
+        """Setting the parameters for the stacking estimator.
+
+        Valid parameter keys can be listed with `get_params()`.
+
+        Parameters
+        ----------
+        params : keyword arguments
+            Specific parameters using e.g.
+            `set_params(parameter_name=new_value)`. In addition, to setting the
+            parameters of the stacking estimator, the individual estimator of
+            the stacking estimators can also be set, or can be removed by
+            setting them to 'drop'.
+
+        Examples
+        --------
+        # In this example, the RandomForestClassifier is removed
+        clf1 = LogisticRegression()
+        clf2 = RandomForestClassifier()
+        eclf = StackingClassifier(estimators=[('lr', clf1), ('rf', clf2)]
+        eclf.set_params(rf='drop')
+        """
+        super()._set_params('estimators', **params)
+        return self
+
+    def get_params(self, deep=True):
+        """Get the parameters of the stacking estimator.
+
+        Parameters
+        ----------
+        deep : bool
+            Setting it to True gets the various classifiers and the parameters
+            of the classifiers as well.
+        """
+        return super()._get_params('estimators', deep=deep)
+
+    def _concatenate_predictions(self, predictions):
+        """Concatenate the predictions of each first layer learner.
+
+        This helper is in charge of ensuring the preditions are 2D arrays and
+        it will drop one of the probability column when the problem is binary
+        since a p(c=1) = 1 - p(c=2).
+        """
+        X_meta = []
+        for est_idx, preds in enumerate(predictions):
+            # case where the the estimator returned a 1D array
+            if preds.ndim == 1:
+                X_meta.append(preds.reshape(-1, 1))
+            else:
+                # remove one column in case of probabilities with binary case
+                if (self.predict_method_[est_idx] == 'predict_proba' and
+                        preds.shape[1] == 2):
+                    X_meta.append(preds[:, [1]])
```

Comment:
This behavior makes sense (and was what I was expecting) but I didn't see it documented anywhere.